src/Command/AbsenceImportCSVCommand.php

```diff
@@ -351,47 +355,64 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             }
             $absence = $absences[$uid];
 
-            if ($absence['perso_id'] != $perso_id
-                or $absence['debut'] != $debut
-                or $absence['fin'] != $fin
-                or $absence['commentaires'] != $commentaires
-                or $absence['valide_n1'] != $valide_n1
-                or $absence['valide'] != $valide_n2) {
+            if ($absence->getUser() != $perso_id
+                or $absence->getStart() != $debut
+                or $absence->getEnd() != $fin
+                or $absence->getComment() != $commentaires
+                or $absence->getValidLevel1() != $valide_n1
+                or $absence->getValidLevel2() != $valide_n2) {
                 // Si l'absence a changé, on met à jour la base de données
                 if ($debug) {
                     $this->log("Si l'absence a changé, on met à jour la base de données", 'AbsenceImportCSV');
                 }
-                $update = array(':perso_id' => $perso_id, ':debut' => $debut, ':fin' => $fin, ':commentaires' => $commentaires, ':valide' => $valide_n2, ':validation' => $validation_n2, ':valide_n1' => $valide_n1, ':validation_n1' => $validation_n1, ':id' => $absence['id']);
-
-            $dbu->execute($update);
+                $update = array('perso_id' => $perso_id, 'debut' => $debut, 'fin' => $fin, 'commentaires' => $commentaires, 'valide' => $valide_n2, 'validation' => $validation_n2, 'valide_n1' => $valide_n1, 'validation_n1' => $validation_n1, 'id' => $absence['id']);
+
+                $absu = $this->entityManager->getRepository(Absence::class)->find($update['id']);
+
+                $absu->setStart($update['debut']);
+                $absu->setEnd($update['fin']);
+                $absu->setRequestDate($update['demande']);
+                $absu->setValidLevel2Date($update['validation']);
+                $absu->setValidLevel1Date($update['validation_n1']);
+                $absu->setUser($update['perso_id'] ?? null);
+                $absu->setValidLevel2($update['valide'] ?? 0);
+                $absu->setValidLevel1($update['valide_n1'] ?? 0);
+                $absu->setReason($update['motif'] ?? '');
+                $absu->setComment($update['commentaires'] ?? '');
+                $absu->setCalName('hamac');
+                $absu->setICalKey($update['ical_key'] ?? null);
+                $absu->setUid($update['uid'] ?? null);
+                $absu->setOtherReason($update['motif_autre'] ?? '');
+                $absu->setStatus($update['etat'] ?? '');
+                $absu->setOriginId($update['id_origin'] ?? 0);
```

Comment:
Nous n'avons plus besoin du tableau $update. Supprime-le, ensuite tu utiliseras directement les variables.

Attention, en ligne 368, tu utilises `$absence['id']`
Comme plus haut, tu fais `$absence = $absences[$uid]; ` , $absence est un objet (entitité), pas un tableau.
Tu dois alors utiliser `$absence->getId()`
^ Je te laisse vérifier ceci.

Si c'est le cas, tu n'as pas besoin de faire `$absu = $this->entityManager->getRepository(Absence::class)->find($update['id']);`
car tu as déjà ton entitié à modifier dans $absence.

Il ne reste donc plus qu'à faire 
```
$absence->setStart($start);
$absence->setEnd($end);
$absence->setRequestDate($requestDate);
...
```